auto-mode: init fails to create tasks/ directory when slice already exists

## Problem
When auto-mode dispatches a plan-slice unit for an existing slice directory, the tasks/ subdirectory is not created, causing a doctor error that blocks execution.

## Environment
- GSD version: 2.25.0
- Unit: plan-slice M001/S01

## Reproduction Context
1. User ran `/gsd` to initialize milestone M001 and slice S01 (CONTEXT.md created)
2. Slice directory S01/ exists with S01-PLAN.md
3. User ran `/gsd auto` to start auto-mode
4. System dispatched plan-slice for M001/S01
5. `ensurePreconditions()` was called but tasks/ directory was not created
6. Doctor check subsequently flagged: `[ERROR] M001/S01 | missing_tasks_dir`

## Forensic Findings
- Anomaly: Missing tasks directory for M001/S01 (Code: missing_tasks_dir)
- Scope: slice
- File: .gsd/milestones/M001/slices/S01
- Fixable: yes (doctor can create it)

The issue is in `ensurePreconditions()` which has two code paths for creating the tasks/ directory:
1. When slice directory is newly created (mkdirSync with recursive creates both)
2. When slice directory already exists (conditional mkdirSync for tasks/ only)

The second path appears to have a condition that doesn't trigger properly when the slice exists but tasks/ does not.

## Suggested Fix Area
File: auto.ts
Function: `ensurePreconditions()` (lines ~3319-3355)

The logic at lines 3345-3351:
```typescript
} else {
  // Ensure tasks/ subdir exists
  const tasksDir = join(slicesDir, sDir, "tasks");
  if (!existsSync(tasksDir)) {
    mkdirSync(tasksDir, { recursive: true });
  }
}
```

Should verify this path is correctly executed when:
- `sDir` is resolved from an existing slice directory
- The tasks/ subdirectory does not yet exist

---
*Auto-generated by `/gsd forensics`*

--- a/src/resources/extensions/gsd/auto.ts
+++ b/src/resources/extensions/gsd/auto.ts
@@ -3309,15 +3309,16 @@ function ensurePreconditions(
       const slicesDir = join(mDirResolved, "slices");
       const sDir = resolveDir(slicesDir, sid);
       if (!sDir) {
-        // Create slice dir with bare ID
-        const newSliceDir = join(slicesDir, sid);
-        mkdirSync(join(newSliceDir, "tasks"), { recursive: true });
-      } else {
-        // Ensure tasks/ subdir exists
-        const tasksDir = join(slicesDir, sDir, "tasks");
-        if (!existsSync(tasksDir)) {
-          mkdirSync(tasksDir, { recursive: true });
-        }
+        // Create slice dir with bare ID (tasks/ included)
+        mkdirSync(join(slicesDir, sid, "tasks"), { recursive: true });
+      }
+      // Always ensure tasks/ subdir exists — even when slice dir was already
+      // present. Handles the case where a slice was created manually or by a
+      // previous run that didn't create tasks/. (#900)
+      const resolvedSliceDir = resolveDir(slicesDir, sid) ?? sid;
+      const tasksDir = join(slicesDir, resolvedSliceDir, "tasks");
+      if (!existsSync(tasksDir)) {
+        mkdirSync(tasksDir, { recursive: true });
       }
     }
   }

